packages/code-studio/src/log/LogExport.ts

```diff
@@ -61,41 +100,46 @@ function makeSafeToStringify(
   return output;
 }
 
-function getReduxDataString(): string {
+function getReduxDataString(blacklist: string[][]): string {
   const reduxData = store.getState();
   return JSON.stringify(
-    makeSafeToStringify(reduxData),
-    null,
+    makeSafeToStringify(reduxData, blacklist),
+    stringifyReplacer(blacklist),
     2 // Indent w/ 2 spaces
   );
 }
 
-function getMetadata(meta?: Record<string, unknown>): string {
+function getMetadata(
+  blacklist: string[][],
+  meta?: Record<string, unknown>
+): string {
   const metadata = {
     uiVersion: import.meta.env.npm_package_version,
     userAgent: navigator.userAgent,
     ...meta,
   };
 
-  return JSON.stringify(metadata, null, 2);
+  return JSON.stringify(metadata, stringifyReplacer(blacklist), 2);
 }
 
 /**
  * Export support logs with the given name.
  * @param fileNamePrefix The zip file name without the .zip extension. Ex: test will be saved as test.zip
+ * @param blacklist List of JSON paths to blacklist. A JSON path is a list representing the path to that value (e.g. client.data would be `['client', 'data']`)
```

Comment:
Add the JSDoc for `metadata` param as well (it was already missing).
It's just any extra metadata you want to export along with the archive, will be saved as a `metadata.json` file in the archive.

Should also add a `@returns` here - returns a promise that resolves successfully if the log archive is created and downloaded successfully, rejected if there's an error.